pyramid_oereb/lib/records/documents.py

```diff
@@ -6,317 +6,115 @@
+            published_until (datetime.date): Date until when this document is published.
             text_at_web (dict of uri): The multilingual URI to the documents content.
-            official_title (dict of unicode): The official title of the document (multilingual).
             abbreviation (dict of unicode): Short term for this document (multilingual).
-            official_number (unicode): The official number for identification of this document.
-            canton (unicode): The cantonal short term (length of two, like 'NE' or 'BL')
-            municipality (unicode): The code for the municipality.
+            official_number (dict of unicode): The official number for identification of this document.
+            only_in_municipality (int): Restrict document to a specific municipality by code.
             article_numbers (list of unicode): Pointers to specific articles.
             file (bytes): The binary content of the document.
-            articles (list of ArticleRecord): The linked articles.
-            references (list of DocumentRecord): The references to other documents.
         """
-        super(DocumentRecord, self).__init__(law_status, published_from, text_at_web)
 
-        if document_type != u'LegalProvision' and document_type != u'Law' and document_type != u'Hint':
-            raise AttributeError('wrong value for document typ was delivered. Only "LegalProvision", '
-                                 '"Law" and "Hint" are allowed. Value was {0}'.format(document_type))
+        if document_type not in ['Rechtsvorschrift', 'GesetzlicheGrundlage', 'Hinweis']:
+            raise AttributeError('Wrong value for document type was delivered. Only "Rechtsvorschrift", '
+                                 '"GesetzlicheGrundlage" and "Hinweis" are allowed. '
+                                 'Value was {0}'.format(document_type))
         if not isinstance(title, dict):
             warnings.warn('Type of "title" should be "dict"')
-        if official_title and not isinstance(official_title, dict):
-            warnings.warn('Type of "official_title" should be "dict"')
         if abbreviation and not isinstance(abbreviation, dict):
             warnings.warn('Type of "abbreviation" should be "dict"')
+        if official_number and not isinstance(official_number, dict):
+            warnings.warn('Type of "official_number" should be "dict"')
+        if published_from and not isinstance(published_from, datetime.date):
+            warnings.warn('Type of "published_from" should be "datetime.date", not '
+                          + str(type(published_from)))
+        if text_at_web and not isinstance(text_at_web, dict):
+            warnings.warn('Type of "text_at_web" should be "dict"')
         if published_from and not isinstance(published_from, datetime.date):
             warnings.warn('Type of "published_from" should be "datetime.date", not '
                           + str(type(published_from)))
 
```

Comment:
shouldn't we test here for the type of the new attributes `only_in_municipality`, `article_nb` and `published_until`?

Reply:
This might be a good completion :D

Reply:
Done 😉